core/models.py

```diff
@@ -1207,12 +1204,14 @@ def __unicode__(self):
         return "path=%s size=%s sha1=%s" % (self.path, self.size, self.sha1)
 
     def _add_page(self, page, tar):
+        from index import get_page_text
+
         d = page.issue.date_issued
         relative_dir = "%s/%i/%02i/%02i/ed-%i/seq-%i/" % (page.issue.title_id, d.year, d.month, d.day, page.issue.edition, page.sequence)
 
         # add ocr text
         txt_filename = relative_dir + "ocr.txt"
-        ocr_text = page.ocr.text.encode('utf-8')
+        ocr_text = get_page_text(page)[0].encode('utf-8')
```

Comment:
How do you know that `get_page_text(page)` will be populated and not throw an array index exception?

Reply:
New implementation of get_page_text precludes returning of an empty array. It now returns ['Text not available'] even if there is no ocr for the page. If you see that it's not the case please let me know where you think the function can return an empty array or None. In the meantime I am going to add a length check condition (in core/models.py line 1214) on results of get_page_text(page) to make it even safer